Game/Scripts/UIOptionsMenu.cpp

```diff
@@ -1,62 +1,693 @@
+					sliderLoad->ModifyCurrentValue(buttonInfo.actualOption);
+				}
+			}
+		}
+		actualConfig.push_back(canvasInfo);
+	}
+
+	// THE CANVAS CONTROLLER ONLY HAVE IMG INSIDE AND DONT SAVE BUTTONS - MADE AN ERROR ON THE SAVE AND LOAD.
+	CanvasOptionInfo controller;
+	actualConfig.push_back(controller);
+
+	for (int a = 0; a < actualConfig.size(); a++)
+	{
+		for (int b = 0; b < actualConfig[a].options.size(); b++)
+		{
+			ApplyChanges(a, b, actualConfig[a].options[b].actualOption);
+		}
+	}
+}
+
+void UIOptionsMenu::LoadDefaultOptions()
+{
+	for (int a = 0; a < buttonsAndCanvas.size()-1; a++)
+	{
+		for (int b = 0; b < actualConfig[a].options.size(); b++)
+		{
+			if (!IsSlider(a, b, 0))
+			{
+				buttonsAndCanvas[a].canvas->GetChildren()[b]->GetChildren()[1]->
+					GetChildren()[actualConfig[a].options[b].actualOption]->Disable();
+				buttonsAndCanvas[a].canvas->GetChildren()[b]->GetChildren()[1]->
+					GetChildren()[actualConfig[a].options[b].defaultOption]->Enable();
+			}
+			else
+			{
+				ComponentSlider* sliderLoad;
+				sliderLoad = buttonsAndCanvas[a].canvas->GetChildren()[b]->
+					GetChildren()[1]->GetChildren()[0]->GetComponent<ComponentSlider>();
+				sliderLoad->ModifyCurrentValue(actualConfig[a].options[b].defaultOption);
+			}
+
+			actualConfig[a].options[b].actualOption = actualConfig[a].options[b].defaultOption;
+			ApplyChanges(a, b, actualConfig[a].options[b].defaultOption);
+		}
+	}
+}
+
+void UIOptionsMenu::SaveOptions()
+{
+
+	std::string optionMenuPath = "Settings/OptionsConfig.txt";
+	rapidjson::Document doc;
+	Json optionsMenu(doc, doc);
+	ModuleFileSystem* fileSystem = App->GetModule<ModuleFileSystem>();
+
+	for (int a = 0; a < actualConfig.size()-1; a++)
+	{
+		Json canvas = optionsMenu[a];
+		for (int b = 0; b < actualConfig[a].options.size(); b++)
+		{
+			canvas[b]["Actual_Option"] = actualConfig[a].options[b].actualOption;
+			canvas[b]["Default_Option"] = actualConfig[a].options[b].defaultOption;
+			canvas[b]["Locked_Option"] = actualConfig[a].options[b].locked;
+		}
+	}
+
+	rapidjson::StringBuffer buffer;
+	optionsMenu.toBuffer(buffer);
+	fileSystem->Save(optionMenuPath.c_str(), buffer.GetString(), (unsigned int)buffer.GetSize());
+}
+
+void UIOptionsMenu::BackToLastSavedOption()
+{
+	if (newSelectedOption == -1)
+	{
+		return;
+	}
+
+	int saveSelectedOption = actualConfig[headerMenuPosition].options[actualButton].actualOption;
+	
+	if (isSlider)
+	{
+		slider->ModifyCurrentValue(saveSelectedOption);
+		isSlider = false;
+	}
+	else
+	{
+		buttonsAndCanvas[headerMenuPosition].canvas->GetChildren()[actualButton]->GetChildren()[1]->
+			GetChildren()[newSelectedOption]->Disable();
+		buttonsAndCanvas[headerMenuPosition].canvas->GetChildren()[actualButton]->GetChildren()[1]->
+			GetChildren()[saveSelectedOption]->Enable();
+	}
+}
+
+bool UIOptionsMenu::IsSlider(int header, int button, int option)
+{
+	if (buttonsAndCanvas[header].canvas->GetChildren()[button]->GetChildren()[1]->GetChildren()[option]->
```

Comment:
you can delete the if-else block and use

```suggestion
return buttonsAndCanvas[header].canvas->GetChildren()[button]->GetChildren()[1]->GetChildren()[option]->HasComponent<ComponentSlider>();
```